subsys/mgmt/mcumgr/lib/cmd/fs_mgmt/src/hash_checksum_mgmt.c

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2018-2021 mcumgr authors
+ * Copyright (c) 2022 Laird Connectivity
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <string.h>
+#include "mgmt/mgmt.h"
+#include "fs_mgmt/hash_checksum_mgmt.h"
+
+static struct hash_checksum_mgmt_group *hash_checksum_mgmt_group_list;
+static struct hash_checksum_mgmt_group *hash_checksum_mgmt_group_list_end;
+
+void hash_checksum_mgmt_unregister_group(struct hash_checksum_mgmt_group *group)
+{
+	struct hash_checksum_mgmt_group *curr = hash_checksum_mgmt_group_list, *prev = NULL;
+
+	if (!group) {
+		return;
+	}
+
+	if (curr == group) {
+		hash_checksum_mgmt_group_list = curr->hc_mg_next;
+		return;
+	}
+
+	while (curr && curr != group) {
+		prev = curr;
+		curr = curr->hc_mg_next;
+	}
+
+	if (!prev || !curr) {
+		return;
+	}
+
+	prev->hc_mg_next = curr->hc_mg_next;
+	if (curr->hc_mg_next == NULL) {
+		hash_checksum_mgmt_group_list_end = curr;
+	}
```

Comment:
You may use slist from Zephyr (https://docs.zephyrproject.org/latest/kernel/data_structures/slist.html) here.

Reply:
This the base code was a copy and paste from the mgmt handler functions, guess we need a bug to update that also

Reply:
FIxed